### PR TITLE
Feat: post zhejiang fixes

### PR DESCRIPTION
**Pull request by TheDZhon**
Follow-up fixes for #482. 

The protocol had been deployed on Zhejiang using commit [`f4260d168029d580919b4625b6e03e0fa983c27a`](https://github.com/lidofinance/lido-dao/tree/f4260d168029d580919b4625b6e03e0fa983c27a).

### CODE REVIEW

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol [hunk 1]
```diff
@@ -512,12 +533,18 @@ contract OracleReportSanityChecker is AccessControlEnumerable {
 
     function _checkFinalizationShareRate(
         LimitsList memory _limitsList,
-        address _lido,
+        uint256 _noWithdrawalsPostTotalPooledEther,
```
Comment (folkyatina) [inline contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:536]: `beforeWithdrawals`?

Comment (folkyatina) [reply contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:536]: Or 'preWithdrawals'?

File: contracts/0.8.9/Burner.sol [hunk 2]
```diff
@@ -54,10 +54,10 @@ contract Burner is IBurner, AccessControlEnumerable {
 
     error ErrorAppAuthLidoFailed();
     error ErrorDirectETHTransfer();
-    error ZeroRecoveryAmount();
-    error StETHRecoveryWrongFunc();
-    error ZeroBurnAmount();
-    error NotEnoughExcessStETH();
+    error ErrorZeroRecoveryAmount();
```
Comment (folkyatina) [inline contracts/0.8.9/Burner.sol:57]: I don't thing that this prefix is meaningful. And also, there is a ton of `Zero...` errors across the codebase.
We need to change them all or revert these changes. I'd rather revert. If you have something reverted it's obviously a error, we don't need to repeat it everywhere.

File: contracts/0.8.9/DepositSecurityModule.sol [hunk 3]
```diff
@@ -49,6 +49,14 @@ contract DepositSecurityModule {
     event DepositsUnpaused(uint24 indexed stakingModuleId);
 
     error ErrorStakingModuleIdTooLarge();
+    error ErrorZeroAddress(string field);
```
Comment (folkyatina) [inline contracts/0.8.9/DepositSecurityModule.sol:53]: I'd rather remove the Error prefix.

File: contracts/0.4.24/Lido.sol [hunk 4]
```diff
@@ -642,7 +642,6 @@ contract Lido is Versioned, StETHPermit, AragonApp {
         require(canDeposit(), "CAN_NOT_DEPOSIT");
 
         IWithdrawalQueue withdrawalQueue = IWithdrawalQueue(locator.withdrawalQueue());
-        require(!withdrawalQueue.isBunkerModeActive(), "CANT_DEPOSIT_IN_BUNKER_MODE");
```
Comment (folkyatina) [inline contracts/0.4.24/Lido.sol:626]: Why so?

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol [hunk 5]
```diff
@@ -453,6 +459,14 @@ contract OracleReportSanityChecker is AccessControlEnumerable {
             revert IncorrectWithdrawalsVaultBalance(_actualWithdrawalVaultBalance);
     }
 
+    function _checkELRewardsVaultBalance(
+        uint256 _actualELRewardsVaultBalance,
+        uint256 _reportedELRewardsVaultBalance
+    ) internal pure {
+        if (_reportedELRewardsVaultBalance > _actualELRewardsVaultBalance)
```
Comment (folkyatina) [inline contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:466]: nitpicking alert. Please, add `{}` to multiline if-clauses.

File: contracts/0.8.9/OracleDaemonConfig.sol [hunk 6]
```diff
@@ -71,6 +72,7 @@ contract OracleDaemonConfig is AccessControlEnumerable {
     }
 
     error ErrorValueExists(string key);
```
Comment (folkyatina) [inline contracts/0.8.9/OracleDaemonConfig.sol:75]: Meaningless `Error` prefix

File: contracts/0.8.9/DepositSecurityModule.sol (hunk 3 above)
Comment (folkyatina) [reply contracts/0.8.9/DepositSecurityModule.sol:53]: e.g `ErrorSignature` even doesn't explain anything. It's better to be `WrongSignature` or `InvalidSignature` or even `ImpossibleToVerifySignature`.

**Review by folkyatina — COMMENTED** [review]
My main concern is about a useless `Error` prefix. It's discussable, but i'd rather prefer to have more meaningful error names.

File: contracts/0.4.24/Lido.sol [hunk 7]
```diff
@@ -181,6 +181,13 @@ contract Lido is Versioned, StETHPermit, AragonApp {
     // Staking limit was removed
     event StakingLimitRemoved();
 
+    // Emits when validators number delivered by the oracle
+    event CLValidatorsUpdated(
+        uint256 indexed reportTimestamp,
```
[the remark this answers is not in the document]
Comment (TheDZhon) [reply contracts/0.4.24/Lido.sol:186]: we do this for the report timestamps in `Lido`

File: contracts/0.4.24/Lido.sol (hunk 4 above)
Comment (TheDZhon) [reply contracts/0.4.24/Lido.sol:626]: it's design decision based on the current bunker mode's derivation

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol (hunk 1 above)
Comment (TheDZhon) [reply contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:536]: it's literally 'withoutWithdrawals`

File: contracts/0.8.9/Burner.sol (hunk 2 above)
Comment (TheDZhon) [reply contracts/0.8.9/Burner.sol:57]: refactoring here and there, thx!

File: contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol (hunk 5 above)
Comment (TheDZhon) [reply contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol:466]: :+1:

File: contracts/0.8.9/OracleDaemonConfig.sol (hunk 6 above)
Comment (TheDZhon) [reply contracts/0.8.9/OracleDaemonConfig.sol:75]: yeah

File: contracts/0.4.24/Lido.sol [hunk 8]
```diff
@@ -1231,6 +1296,42 @@ contract Lido is Versioned, StETHPermit, AragonApp {
             reportContext,
             IPostTokenRebaseReceiver(_contracts.postTokenRebaseReceiver)
         );
+
+        // Step 9. Sanity check for the provided simulated share rate
+        if (_reportedData.lastFinalizableRequestId != DONT_FINALIZE_WITHDRAWALS) {
+            uint256 noWithdrawalsPostTotalShares = postTotalShares;
```
Comment (ujenjt) [inline contracts/0.4.24/Lido.sol:1302]: `postTotalSharesIfNoWithdrawalsProcessed` - ?

File: contracts/0.4.24/StETH.sol [hunk 9]
```diff
@@ -363,8 +363,6 @@ contract StETH is IERC20, Pausable {
     function transferSharesFrom(
         address _sender, address _recipient, uint256 _sharesAmount
     ) external returns (uint256) {
-        _whenNotStopped();
```
Comment (ujenjt) [inline contracts/0.4.24/StETH.sol:366]: _transferShares
